src/views/common/TimeScrollView/YAxisTicks.tsx

```diff
@@ -32,27 +32,65 @@ const range = (min: number, max: number, step: number, base: number) => {
+ */
 const fitGridLines = (minGridLines: number, maxGridLines: number, range: number): {step: number, scale: number} => {
     let scale = 0
     while (true) {
-        const steps = [1, 2, 5]
-        const realizedScale = Math.pow(10, scale)
+        const steps = [1, 2, 5]  // candidate step sizes are 1, 2, 5, (10, 20, 50, 100, 200, ...)
+        const realizedScale = Math.pow(10, scale)   // realizedScale = the base order of magnitude represented by `scale`.
         const results: {step: number, scale: number}[] = []
-        for (let s of steps) {
+        for (let s of steps) {  // for each of the 3 candidate step sizes at this order of magnitude
+            // determine how many steps of [1 | 2 | 5] * 10^scale would be needed to exhaust the range.
+            // (technically on the plot this will be the number of spaces between the grid lines, thus the hard </>).
             const fit = range/(s * realizedScale)
             if (fit > minGridLines && fit < maxGridLines) {
+                // if this split produces an acceptable number of counts, record the step and scale sizes that work.
                 results.push({step: s, scale: scale})
             }
-            // this means the step size is too big. This shouldn't really happen without finding an acceptable step size first,
-            // but we'll check for it later just in case.
-            if (fit < minGridLines) { return {step: -1, scale: -1} }
-            results.push({step: 0, scale: 0})
+            if (fit < minGridLines) {
+                // step size was too big--it resulted in too few divisions of the range.
+                results.push({step: -1, scale: -1})
+                break
+            }
         }
+        // Completing this loop means we finished evaluating one order of magnitude. If we found
+        // a step-scale combination that gave a number of divisions in the acceptable range,
+        // it should show up as a result record with a positive step size (...viz, 1, 2, or 5).
+        // We never have to worry about it actually being from a different scale, because we run
+        // this operation every time we complete an order of magnitude.
         const a = results.find(r => r.step > 0)
         if (a) return a
+        // On the other hand, if the result set includes an entry with a step size of -1, then
+        // the last loop's step size got too big. This is bad--if we see this, it means it
+        // is not productive to try larger step sizes, but if we had found something that did
+        // work, we would've returned that first. So realistically this probably means the
+        // range of allowed grid lines is too small. (Those are set programmatically from
+        // a range of allowable inter-tick pixel counts, minGridSpacingPx and maxGridSpacingPx,
+        // which might need to be adjusted.)
         const b = results.find(r => r.step === -1)
         if (b) return b
```

Comment:
In this case the step size was too large and we are returning `{step: -1, scale: -1}`. The docs for the return value doesn't indicate this case.
Also, don't we want to allow negative values of scale? For example, a range of 0.008 or something. I think there should be a `scale = scale - 1` to try the next smaller order of magnitude.
Then we could check for both cases `scale > 10` and `scale < -10`.

Reply:
HOWEVER, we want to avoid infinite loop at all costs. For example, we don't want to get into a increment/decrement infinite cycle. This can happen depending on the values of minGridLines and maxGridLines. If we make that criteria too narrow, then you'll get the infinite back-and-forth. We need to theorize a bit about this (or use a different mechanism).

Reply:
The input data range is scaled to (100, 1000] in l. 164-166:
```Typescript
        const rangeScale = Math.round(Math.log10(dataRange))
        const zoomedRangeScale = 3 - rangeScale // make sure we're counting through the range with whole numbers
        const zoomedRange = Math.round(dataRange * (Math.pow(10, zoomedRangeScale)))
```
and whatever scale comes out of `fitGridLines` then gets corrected by taking that scaling factor back out (in line 175).

So it shouldn't ever be possible to need a stepsize smaller than 1--practically, I expect the `fitGridSize()` function should almost always be returning something in the 20-200 range, but it will depend on how tight the tolerances are for pixel heights and how squashed you make the graph.

I'll annotate this part more as well.

I think the bigger issue we were running into, practically speaking, is that the loop in `fitGridSize` *returned* the -1 value immediately instead of letting it get picked up in the `while` loop--so since we don't look for a "successful" step size until we're done with checking [1, 2, 5] at that order of magnitude, if stepsize 2 works and stepsize 5 is too big, then I was returning a failure instead of catching that 2 was acceptable.

Reply:
I see. I think that's confusing, but I guess it should work.
So there was a bug in the prior version?

Reply:
Yeah. I corrected it in this PR by changing it to pushing `step: -1` into the results instead of returning it right away.